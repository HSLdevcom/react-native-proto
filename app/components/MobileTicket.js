/**
 * MobileTicket component
 * @flow
 */

import React, {Component} from 'react';
import {Linking, Modal, StyleSheet, Text, TouchableOpacity, View} from 'react-native';
import colors from '../colors';

const styles = StyleSheet.create({
    container: {
        alignItems: 'center',
        flex: 1,
        justifyContent: 'center',
        padding: 15,
    },
    header: {
        fontSize: 30,
    },
    text: {
        color: 'black',
        textAlign: 'center',
    },
    button: {
        backgroundColor: colors.brandColor,
        borderColor: colors.lightGrayBg,
        borderRadius: 10,
        borderWidth: 1,
        marginTop: 10,
        padding: 10,
        width: '90%',
    },
    buttonText: {
        color: 'white',
        textAlign: 'center',
    },
});


class MobileTicket extends Component { // eslint-disable-line react/prefer-stateless-function
    state = {
        showModal: false,
    }

    closeModal = () => this.setState({showModal: false});

    maybeOpenMobileTicket = () => {
        Linking.canOpenURL('com.hsl.mobiililippu:?tt=1111')
        .then((isInstalled) => {
            if (isInstalled) {
                Linking.openURL('com.hsl.mobiililippu:?tt=1111').catch(e => console.log(e));
            } else {
                this.setState({showModal: true});
            }
        });
    }
    render() {
        const modal = this.state.showModal ?
        (
            <Modal
                animationType="slide"
                transparent={false}
                visible
            >
                <View style={{flex: 1, alignItems: 'center', justifyContent: 'center'}}>
                    <Text>HSL Mobiililippu ei ole asennettuna puhelimeesi.</Text>
                    <TouchableOpacity style={styles.button} onPress={this.closeModal}>
                        <Text style={styles.buttonText}>Sulje</Text>
                    </TouchableOpacity>
                </View>
            </Modal>
        ) : null;
        return (
            <View style={styles.container}>
                {modal}
                <Text style={[styles.text, styles.header]}>
                    Osta matkalippu suoraan puhelimeesi
                </Text>
                <Text style={styles.text}>Liput kelpaavat kaikkiin HSL:n kulkuvälineisiin</Text>
                <Text style={styles.text}>
                    Tunnistautumalla saat lippuja yhtä edulliseen
                    hintaan kuin matkakortilla ostettaessa
                </Text>
                <TouchableOpacity style={styles.button} onPress={this.maybeOpenMobileTicket}>
                    <Text style={styles.buttonText}>Kertalippu</Text>
                </TouchableOpacity>
                <TouchableOpacity style={styles.button} onPress={this.maybeOpenMobileTicket}>
                    <Text style={styles.buttonText}>Vuorokausilippu</Text>
                </TouchableOpacity>
                <TouchableOpacity style={styles.button} onPress={this.maybeOpenMobileTicket}>
                    <Text style={styles.buttonText}>Kausilippu</Text>
                </TouchableOpacity>
            </View>
        );
    }
}

export default MobileTicket;
