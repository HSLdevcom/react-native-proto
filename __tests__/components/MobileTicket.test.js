// https://facebook.github.io/jest/blog/2016/07/27/jest-14.html
import 'react-native';
import React from 'react';
// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';
import MobileTicket from '../../app/components/MobileTicket';

it('renders correctly', () => {
    const tree = renderer.create(<MobileTicket />).toJSON();
    expect(tree).toMatchSnapshot();
});
