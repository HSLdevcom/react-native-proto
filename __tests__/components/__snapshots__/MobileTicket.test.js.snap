// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
      "padding": 15,
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "black",
          "textAlign": "center",
        },
        Object {
          "fontSize": 30,
        },
      ]
    }
  >
    Osta matkalippu suoraan puhelimeesi
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "black",
        "textAlign": "center",
      }
    }
  >
    Liput kelpaavat kaikkiin HSL:n kulkuvälineisiin
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "black",
        "textAlign": "center",
      }
    }
  >
    Tunnistautumalla saat lippuja yhtä edulliseen hintaan kuin matkakortilla ostettaessa
  </Text>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    isTVSelectable={true}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "rgb(0, 122, 201)",
        "borderColor": "rgb(239, 242, 244)",
        "borderRadius": 10,
        "borderWidth": 1,
        "marginTop": 10,
        "opacity": 1,
        "padding": 10,
        "width": "90%",
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "white",
          "textAlign": "center",
        }
      }
    >
      Kertalippu
    </Text>
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    isTVSelectable={true}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "rgb(0, 122, 201)",
        "borderColor": "rgb(239, 242, 244)",
        "borderRadius": 10,
        "borderWidth": 1,
        "marginTop": 10,
        "opacity": 1,
        "padding": 10,
        "width": "90%",
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "white",
          "textAlign": "center",
        }
      }
    >
      Vuorokausilippu
    </Text>
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    isTVSelectable={true}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "rgb(0, 122, 201)",
        "borderColor": "rgb(239, 242, 244)",
        "borderRadius": 10,
        "borderWidth": 1,
        "marginTop": 10,
        "opacity": 1,
        "padding": 10,
        "width": "90%",
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "white",
          "textAlign": "center",
        }
      }
    >
      Kausilippu
    </Text>
  </View>
</View>
`;
